source/common/upstream/upstream_impl.cc

```diff
@@ -120,10 +120,20 @@ createProtocolOptionsConfig(const std::string& name, const ProtobufWkt::Any& typ
   Server::Configuration::ProtocolOptionsFactory* factory =
       Registry::FactoryRegistry<Server::Configuration::NamedNetworkFilterConfigFactory>::getFactory(
           name);
+
+  if (Registry::FactoryRegistry<
+          Server::Configuration::NamedNetworkFilterConfigFactory>::isDeprecated(name)) {
+    ENVOY_LOG_MISC(warn, "deprecated network filter config name was used {}", name);
```

Comment:
We should probably have some tests for this with `EXPECT_LOG`.

Reply:
I think that it is ok to test only `isDeprecated()` because `ENVOY_LOG_MISC` called certainly if result is `true`.